### PR TITLE
Add manifest typing and lib for caching in kraken

### PR DESCRIPTION
**Pull request by JoaoMario109**
Needed in #2486

### CODE REVIEW

File: core/services/kraken/manifest/manifest.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    def is_expired(self) -> bool:
+        """
+        Checks if the manifest is expired.
+
+        Returns:
+            bool: True if the manifest is expired, False otherwise.
+        """
+
+        return (int(datetime.datetime.now(datetime.timezone.utc).timestamp()) - self.issued_at) > self.expires_in
+
```
Comment (joaoantoniocardoso) [inline core/services/kraken/manifest/manifest.py:41]: Consider unifying the timestamp into something like:

```Python
@classmethod
def _current_timestamp(cls) -> int:
  return int(datetime.datetime.now(datetime.timezone.utc).timestamp())
```

File: core/services/kraken/manifest/manifest.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        Returns:
+            ManifestContent: Manifest content.
+
+        Raises:
+            RuntimeError: If the manifest could not be fetched.
+        """
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(REPO_URL) as resp:
```
Comment (joaoantoniocardoso) [inline core/services/kraken/manifest/manifest.py:79]: I know it was already implemented, but maybe it's a good idea to add a `, headers={'Accept': 'application/json'}`?

**Review by joaoantoniocardoso — COMMENTED** [review]
Haven't tested yet

File: core/services/kraken/manifest/manifest.py (hunk 1 above)
Comment (JoaoMario109) [reply core/services/kraken/manifest/manifest.py]: Added!

File: core/services/kraken/manifest/manifest.py (hunk 2 above)
Comment (JoaoMario109) [reply core/services/kraken/manifest/manifest.py]: Added!

**Review by patrickelectric — CHANGES_REQUESTED** [review]
@JoaoMario109 can you check what CI is detecting ?

**Comment by patrickelectric** [comment]
merged in https://github.com/bluerobotics/BlueOS/pull/2486

**Pull request closed without merging**